src/Sql/Sql/Elastic Jobs/Cmdlet/PrivateEndpoint/GetAzureSqlElasticJobPrivateEndpoint.cs

```diff
@@ -0,0 +1,170 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Commands.Sql.ElasticJobs.Model;
+using Microsoft.Rest.Azure;
+
+namespace Microsoft.Azure.Commands.Sql.ElasticJobs.Cmdlet.PrivateEndpoint
+{
+    /// <summary>
+    /// Defines the Get-AzSqlElasticJobPrivateEndpoint Cmdlet
+    /// </summary>
+    [Cmdlet(VerbsCommon.Get, ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "SqlElasticJobPrivateEndpoint",
+        DefaultParameterSetName = DefaultParameterSet)]
+    [OutputType(typeof(AzureSqlElasticJobPrivateEndpointModel))]
+    public class GetAzureSqlElasticJobPrivateEndpoint : AzureSqlElasticJobPrivateEndpointCmdletBase<AzureSqlElasticJobAgentModel>
+    {
+        /// <summary>
+        /// Gets or sets the resource group name
+        /// </summary>
+        [Parameter(
+            Mandatory = true,
+            ParameterSetName = DefaultParameterSet,
+            Position = 0,
+            HelpMessage = "The resource group name")]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public override string ResourceGroupName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the agent model input object
+        /// </summary>
+        [Parameter(ParameterSetName = InputObjectParameterSet,
+            Mandatory = true,
+            ValueFromPipeline = true,
+            Position = 0,
+            HelpMessage = "The agent object")]
+        [ValidateNotNullOrEmpty]
+        public AzureSqlElasticJobAgentModel ParentObject { get; set; }
```

Comment:
"ElasticJobAgentObject" could be a more meaningful name

Reply:
I don't mind changing the parameter names but the parent resource AzSqlElasticJobAgent uses ParentObject and ParentResourceId, do you think it would cause any problems for this child resource to be different? 

https://github.com/Azure/azure-powershell/blob/0d7b3d0379ad2792310bfe2f6ad2b329540a1f7e/src/Sql/Sql/Elastic%20Jobs/Cmdlet/Agent/GetAzureSqlElasticJobAgent.cs#L43C8-L64C53

Reply:
I do not recommend to do so in this PR.